bt-core/src/main/java/bt/torrent/messaging/Assignments.java

```diff
@@ -146,7 +146,7 @@ public int workersCount() {
             LOGGER.trace(buf.toString());
         }
 
-        return selectedPiece.isPresent() ? Optional.of(assign(peer, selectedPiece.get())) : Optional.empty();
+        return selectedPiece.map(integer -> assign(peer, integer));
```

Comment:
This class is redesigned in https://github.com/atomashpolskiy/bt/pull/124 and included in https://github.com/atomashpolskiy/bt/pull/126 (the latter is an ongoing attempt on eliminating block data from Java heap)